tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Program.cs

```diff
@@ -196,6 +201,45 @@ public async Task<ExitCondition> RunAsync(
                         {
                             OpenBrowser(definition.GetWebUrl());
                         }
+
+                        updatedDefinitions.Add(definition);
+                    }
+                }
+
+                var serviceConnectionObjects = await context.GetServiceConnectionsAsync(serviceConnections, cancellationToken);
+
+                foreach (var serviceConnection in serviceConnectionObjects)
+                {
+                    // Get set of permissions for the service connection
+                    JsonNode pipelinePermissions = await context.GetPipelinePermissionsAsync(serviceConnection.Id, cancellationToken);
+
+                    var pipelines = pipelinePermissions["pipelines"].AsArray();
+                    var pipelineIdsWithPermissions = new HashSet<int>(pipelines.Select(p => p["id"].GetValue<int>()));
+
+                    bool needsUpdate = false;
+                    foreach (var definition in updatedDefinitions)
```

Comment:
Does this mean if we accidentally run the generator against public with the service connection flag, it will add those pipelines to the permissions list? Or am I missing the logic somewhere else that excludes that. If not, I think it would be better to have the convention forced under the hood to prevent client level mistakes.

Reply:
The service connection lookup is also attached to the team so it would fail for public to find those service connections. However, even if we did find them it should fail to try to add a pipeline from another team project.